### ISSUE TITLE
regression in digdug 2.2.4

### ISSUE THREAD

**Issue by vladikoff** [theintern/digdug#75]
It seems like the recent dot release changed something where with Firefox 58 testing we now get:

```
Suite firefox on linux 4.15.0-1035-aws - signin token code FAILED
InvalidArgument: [POST http://localhost:4444/wd/hub/session/b36ca15a-1246-44de-80b4-69f012454baf/window / {"name":"2147483683"}] missing field `handle` at line 3 column 1
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'
System info: host: '356e9606de2d', ip: '192.168.80.3', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-1035-aws', java.version: '1.8.0_212'
Driver info: driver.version: unknown
  at Command.<anonymous>  <tests/functional/lib/helpers.js:402:12>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:399:6>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:60:23>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:73:18>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:68:10>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:443:6>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:60:23>
  at Command.<anonymous>  <tests/functional/lib/helpers.js:68:10>
  at Suite.beforeEach  <tests/functional/sign_in_token_code.js:51:8>
  at <src/lib/Suite.ts:383:26>
  at runLifecycleMethod  <src/lib/Suite.ts:351:13>
  at next  <src/lib/Suite.ts:522:32>
``` 

We are gonna try to update firefox for this case to see if that fixes it

**Comment by vladikoff** [comment]
Sorry @jason0x43 this is probably a leadfoot issue?  (might have to transfer it)

Shrinkwrap update below https://github.com/mozilla/fxa/pull/1343/files#diff-e860b3ada447521466d04927113e18b0R842

**Comment by jason0x43** [comment]
Yeah, that'd would be a leadfoot issue, unless it's happening because of an update to Selenium and geckodriver. Are you using digdug's defaults, or does your test config pin the driver versions?

**Comment by vladikoff** [comment]
Pretty much defaults
https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/tests/intern.js#L71

**Comment by jason0x43** [comment]
Dig Dug tracks (roughly) the current releases of Selenium and web drivers, so it possible that a Dig Dug update is using a geckodriver that no longer works right with FF 58; or it could be some recent changes to feature tests in Leadfoot. I'll take a look.

**Comment by jason0x43** [comment]
I have so far been unable to reproduce the error; I can't get an old version of FF installed locally, and FF 58 on BrowserStack can use the `name` parameter even when I specify geckodriver 0.24.0.

To test if it's a driver version issue, you could try pinning your Selenium and geckodriver versions to the previous Dig Dug defaults:

```
"tunnelOptions": {
  "version": "3.14.0",
  "drivers": [
    { "name": "firefox", "version": "0.21.0" }
  ]
}
```

**Comment by vladikoff** [comment]
Actually, we are using Firefox 67 for this. 

The API for window probably changed 

 [POST http://localhost:4444/wd/hub/session/97fd0cbf-ab21-2945-bf45-507276f6492b/window / {"name":"4294967336"}] missing field `handle` at line 3 column 1

and now needs to pass in the handle instead of `name` . I've seen this happen before

**Comment by vladikoff** [comment]
@jason0x43 so I looked at this more, this is clearly the `handle` param

I tried setting this here:

```diff
config.capabilities = {
  usesHandleParameter: true,
  usesWebDriverWindowHandleCommands: true,
};
```

But it still logs as `false`. Am I doing it wrong?

https://github.com/mozilla/fxa/commit/78e97429f2bfbef3897ec9b1f9f05c9546ed5869#diff-c2a4b4f157997a366f725852ddefbd1aR123

**Comment by jason0x43** [comment]
You're actually hitting a couple of Leadfoot issues there. One is that the feature test for `usesHandleParameter` is failing on Firefox because the string case in the error doesn't match what the test is looking for (theintern/leadfoot#178).

The other issue is that manually specifying capabilities only works for some drivers right now, and geckodriver apparently isn't one of those (theintern/leadfoot#137).

I should have a fix out for theintern/leadfoot#178 today, and _possibly_ for theintern/leadfoot#137.

**Comment by vladikoff** [comment]
@jason0x43 could you please push the latest leadfoot to npm?

**Comment by jason0x43** [comment]
2.2.12 is now out and includes the fix for theintern/leadfoot#137

**Comment by jason0x43** [comment]
I believe this has been resolved, but we can reopen if not.
